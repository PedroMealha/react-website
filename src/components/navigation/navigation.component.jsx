import { Fragment } from "react";

import NavButton from "../../components/nav-button/nav-button.component";

import './navigation.styles.scss';

const Navigation = () => {

	return (
		<Fragment>
			<div className="navigation">
				<div className="nav-external">
					<NavButton buttonType='about' />
					<NavButton buttonType='linkedin' />
					<NavButton buttonType='github' />
					<NavButton buttonType='behance' />
					<NavButton buttonType='contact' />
				</div>
			</div>
		</Fragment >
	)
}

export default Navigation;