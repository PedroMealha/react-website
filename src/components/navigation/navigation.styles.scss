@import '../../components/global/global.styles.scss';

.navigation {
	display        : flex;
	align-items    : center;
	justify-content: center;
	background     : lighten($color-grey, 6%);
	padding        : 0 20px;

	.nav-external {
		display        : flex;
		flex-direction : column;
		align-items    : center;
		justify-content: center;
	}
}

@media screen and (max-device-width: 768px) {
	.navigation {
		padding: 0 10px;

		.nav-external {
			flex-direction: row;
		}
	}
}