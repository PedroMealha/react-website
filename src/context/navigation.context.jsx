import { createContext, useState } from 'react';

export const NavigationContext = createContext({
	isNavOpen: false
});

export const NavigationProvider = ({ children }) => {
	const [isNavOpen, setIsNavOpen] = useState(false);
	
	const value = { isNavOpen, setIsNavOpen };
	return <NavigationContext.Provider value={ value }>{ children }</NavigationContext.Provider>;
};
