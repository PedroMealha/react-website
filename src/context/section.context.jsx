import { createContext, useReducer } from "react";

const SECTION_TYPES = {
	SET_SECTION: 'SET_SECTION'
};

const INITIAL_SECTION = {
	currentSection: 'profile'
}

const sectionReducer = (state, action) => {
	const { type, payload } = action;

	switch (type) {
		case SECTION_TYPES.SET_SECTION:
			return {
				...{ state },
				currentSection: payload
			}
		default:
			throw new Error(`Unhandled type ${type} in SectionReducer`)
	}
}

export const SectionContext = createContext({
	currentSection: '',
	setCurrentSection: () => { }
})

export const SectionProvider = ({ children }) => {
	const [{ currentSection }, dispatch] = useReducer(sectionReducer, INITIAL_SECTION);

	const setCurrentSection = section => {
		dispatch({
			type: SECTION_TYPES.SET_SECTION,
			payload: section
		})
	}

	const value = { currentSection, setCurrentSection };

	return <SectionContext.Provider value={ value }>{ children }</SectionContext.Provider>
}