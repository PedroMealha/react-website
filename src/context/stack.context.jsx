import { createContext, useState, useEffect } from "react";

import { getStackFromFirestore } from "../utils/firebase/firebase.utils";

export const StackContext = createContext({
	stackMap: {}
})

export const StackProvider = ({ children }) => {
	const [stackMap, setStackMap] = useState({});

	useEffect(() => {
		const getStackMap = async () => {
			const skillMap = await getStackFromFirestore();
			setStackMap(skillMap);
		}

		getStackMap();
	}, []);

	const value = { stackMap };

	return (
		<StackContext.Provider value={ value }>{ children }</StackContext.Provider>
	)
}
