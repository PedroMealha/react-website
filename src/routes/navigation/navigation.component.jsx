import { Fragment } from "react";
import { Outlet } from "react-router-dom";

import NavButton from "../../components/nav-button/nav-button.component";

import './navigation.styles.scss';

const Navigation = () => {

	return (
		<Fragment>
			<div className="navigation">
				<div className="nav-external">
					<NavButton buttonType='profile' />
					<NavButton buttonType='skills' />
					<NavButton buttonType='linkedin' />
					<NavButton buttonType='github' />
					<NavButton buttonType='behance' />
					<NavButton buttonType='contact' />
				</div>
			</div>
			<Outlet />
		</Fragment >
	)
}

export default Navigation;