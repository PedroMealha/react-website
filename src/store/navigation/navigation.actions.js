import { NAV_ACTION_TYPES } from "./navigation.types";
import { createAction } from "../../utils/reducer/reducer.utils";

export const setIsNavOpen = bool =>
	createAction(NAV_ACTION_TYPES.SET_CURRENT_NAV, bool);
