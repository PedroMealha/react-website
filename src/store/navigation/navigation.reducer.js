import { NAV_ACTION_TYPES } from "./navigation.types";

const INITIAL_STATE = {
  isNavOpen: false,
};

export const navigationReducer = (
	state = INITIAL_STATE,
	action = {}
) => {
	const { type, payload } = action;

  switch (type) {
    case NAV_ACTION_TYPES.SET_CURRENT_NAV:
      return { ...state, isNavOpen: payload };
    default:
      return state;
  }
};
