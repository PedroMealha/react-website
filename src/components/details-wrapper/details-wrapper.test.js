import React from "react";
import { shallow } from "enzyme";
import DetailsWrapper from "./details-wrapper.component";
import { NavigationContext } from '../../context/navigation.context';

describe('DetailsWrapper component', () => {
	const wrapper = shallow(<DetailsWrapper />);

	it('expect to render DetailsWrapper component', () => {
		expect(wrapper).toMatchSnapshot();
	})

	it('expect isNavOpen to start as false', () => {
		expect(wrapper.find("isNavOpen").exists()).toBeFalsy();
	})

	it("expect isNavOpen=true to add 'open' class", () => {
		jest.spyOn(React, 'useContext').mockImplementation(() => ({
			isNavOpen: true
		}));

		const ContextComponent = () => (
			<NavigationContext.Provider>
				<DetailsWrapper />
			</NavigationContext.Provider>
		);

		const wrapper = shallow(<ContextComponent />);
		expect(wrapper.dive().dive().find('.app-details-wrapper').hasClass('open')).toBeTruthy();
	});
})