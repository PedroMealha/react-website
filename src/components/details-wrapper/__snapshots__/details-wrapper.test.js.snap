// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsWrapper component expect to render DetailsWrapper component 1`] = `
<div
  className="app-details-wrapper "
>
  <AvatarWrapper />
  <PersonalInfoWrapper />
</div>
`;
